### ISSUE TITLE
Feature: Automated Chocolatey/Winget installer

### ISSUE THREAD

**Issue by andremmfaria** [ferdium/ferdium-app#194]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

Disclaimer: I followed everything that happened with the other app and i am truly sorry for that. This application is amazing and it is a true pity that things turned out the way they did.

That said, I am one of the people who manage their systems using cli applications. I personally use [chocolatey windows package manager](https://chocolatey.org/). But there is also [winget](https://github.com/microsoft/winget-cli) that is growing quite a bit.

Since the other application has both [chocolatey package](https://community.chocolatey.org/packages/ferdi) and a [winget package](https://winget.run/pkg/AmineMouafik/Ferdi) i think this one should have those too. 

Also, as an added bonus, 
### Proposed Solution

Create the Ferdium package on chocolatey using their proposed course
* https://community.chocolatey.org/courses/creating-chocolatey-packages

Create the Ferdium package on winget using their documentation: 
* https://docs.microsoft.com/en-us/windows/package-manager/package

### Alternatives Considered

I understand that we already have scoop but providing the chocolatey and/or winget options might be a good idea too.

**Comment by andremmfaria** [comment]
References:

* Chocolatey: https://github.com/digitalcoyote/chocolatey-packages/tree/master/ferdi
* Winget: https://github.com/microsoft/winget-pkgs/tree/master/manifests/a/AmineMouafik/Ferdi

**Comment by mattbarreiro** [comment]
+1 for winget. If it's just a matter of  CI/CD effort, I would be happy to help

**Comment by mattbarreiro** [comment]
I did some digging, Microsoft uses Actions to automatically open a PR to the WinGet community repo whenever they release a new version of PowerToys. Maybe we could do something similar for Ferdium.

https://github.com/microsoft/PowerToys/blob/main/.github/workflows/package-submissions.yml

**Comment by digitalcoyote** [comment]
I finally got around to setting up the ferdium package. Unfortunately it won't release older packages automatically, but the latest nightly should be available after moderation. Once a stable release is released here, it'll start showing up in general searches.

https://community.chocolatey.org/packages/ferdium Please give it a review and make sure everything is correct. I used a copy of the package scripts I wrote for Ferdi to get it started, so it's possible there are references I missed. Also, until the package is marked as Trusted, there will be a significant delay between github releases and the package being publicly available on Chocolatey.

If anyone wants to test it directly, you can access it now by specifying the version when you install it (caution: this is normally unadvisable when installing packages from Chocolatey since it bypasses some of their tests/protections). Otherwise once it passes automated tests it will be available with the prerelease flag.

**Comment by Alphrag** [comment]
The winget package has also now been created as `ferdium`. This contains nightly for now but will be used for the official release when we make it and then we'll create an additional `ferdium-nightly` for those that want to keep up on the beta versions.
It is not yet automated, as I need to fiddle a bit on how to automatically parse the json of the release to find the urls of the installers as this is the last missing step for the github action to exist. So should be good hopefully by the end of the week!

**Comment by vraravam** [comment]
On further thought, I would recommend that we start off with the `ferdium-nightly` from the get-go itself (and not rename at a later point).
1. It'll give a clear message to the users that this is a **nightly**
2. For nightly users, there wont be a shift when we do end up releasing the betas/stable versions.

**Comment by russellbanks** [comment]
On WinGet, the package version in the manifest needs to match the version that Control Panel shows for Ferdium, which is `6.0.0.0`. This means that even though Ferdium will be having nightly updates, WinGet won't alert the user to an update because the version on Control Panel is not changing. I would suggest that the Ferdium nightly versions have different version numbers in Control Panel in order to mitigate this.

**Comment by russellbanks** [comment]
> It is not yet automated, as I need to fiddle a bit on how to automatically parse the json of the release to find the urls of the installers as this is the last missing step for the github action to exist. So should be good hopefully by the end of the week!

You can use [WinGet Releaser](https://github.com/vedantmgoyal2009/winget-releaser) for this. It's a workflow that runs on release, automatically creates the updated manifest, and makes a pull request to winget-pkgs with it.

**Comment by SpecialAro** [comment]
> On WinGet, the package version in the manifest needs to match the version that Control Panel shows for Ferdium, which is `6.0.0.0`. This means that even though Ferdium will be having nightly updates, WinGet won't alert the user to an update because the version on Control Panel is not changing. I would suggest that the Ferdium nightly versions have different version numbers in Control Panel in order to mitigate this.

@Alphrag this is due to the fact we're having the `msi` instead of the `exe` installer right?

@russellbanks if it is as I'm saying we're moving this, so it will be fixed :D

**Comment by Alphrag** [comment]
> @Alphrag this is due to the fact we're having the `msi` instead of the `exe` installer right?

Yes exactly, that is a problem coming from the `msi` installer as we only use 3 levels for versioning and then an additional reference for pre-release as recommended in the official notice for [semantic versioning (point 9) ](https://semver.org/#spec-item-9). So this is not our doing and the version on winget follows our current versioning system, but when using the `msi` installer it does something wrong. We are going to dump this and only keep the `nsis` installer (that corresponds to the `AutoSetup*.exe`) from now on until such an issue is fixed. I only noticed the extend of this problem yesterday so this is latent.


> You can use [WinGet Releaser](https://github.com/vedantmgoyal2009/winget-releaser) for this. It's a workflow that runs on release, automatically creates the updated manifest, and makes a pull request to winget-pkgs with it.

Thanks for the link, his action looks nice but I decided to write my own version (using similar templates), allowing us to bump it in different ways (and also taking care of multiple arch if needed). I will also change the current `ferdium` winget manifest, that will only contain the stable version, and will create new `ferdium-beta` and `ferdium-nightly` for the different level of pre-releases. The script is now finished and tested (it was fiddly) but needs first the setup of the new winget manifests and the correction of the previous one. All in all, you should expect it to be ready by the end of the weekend :wink:
